app/services/cloud_topology_service.rb

```diff
@@ -0,0 +1,77 @@
+class CloudTopologyService < TopologyService
+  include UiServiceMixin
+
+  @provider_class = ManageIQ::Providers::CloudManager
+
+  def entity_type(entity)
+    entity.class.name.demodulize
+  end
+
+  def build_topology
+    topo_items = {}
+    links = []
+
+    included_relations = [
+      :tags,
+      :availability_zones => :tags,
+      :cloud_tenants      => [
+        :tags,
+        :vms => :tags,
+      ],
+    ]
```

Comment:
This should really look like

```
included_relations = [
  :tags,
  {:availability_zones => :tags,
   :cloud_tenants => [:tags, {:vms=>:tags}]}
]
```

Otherwise, it looks like a weird mixture of array and hash, and does not really convey that both `:cloud_tenants` and `:availability_zones` together form the second entry in the array.